include/openenclave/internal/str.h

```diff
@@ -424,6 +424,78 @@ MEM_INLINE int str_split(str_t* str, const char* delim, str_t* lhs, str_t* rhs)
     return 0;
 }
 
+/**
+ * This type contains a Universally Unique Resource Identifier (UUID) type as
+ * defined in RFC4122. These UUID values are used to identify Trusted
+ * Applications.
+ */
+typedef struct
+{
+    uint32_t timeLow;
+    uint16_t timeMid;
+    uint16_t timeHiAndVersion;
+    uint8_t clockSeqAndNode[8];
+} TEEC_UUID;
+
+MEM_INLINE int uuid_from_string(str_t* str, TEEC_UUID* uuid)
+{
+    int i;
+    uint64_t uuid_parts[5];
+    char* id_copy;
+    char* posn;
+    char* current_token;
+
+#if defined(__linux__)
+    id_copy = strdup(str_ptr(str));
+#else
+    id_copy = _strdup(str_ptr(str));
+#endif
+    /* Remove ".ta" extension, if one is present */
+    size_t len = strlen(id_copy);
+    if ((len > 3) && (strcmp(&id_copy[len - 3], ".ta") == 0))
+    {
+        id_copy[len - 3] = 0;
+    }
+
+    if (strlen(id_copy) != 36)
+        return -1;
+
+    i = 5;
+#if defined(__linux__)
+    current_token = strtok_r(id_copy, "-", &posn);
+#else
+    current_token = strtok_s(id_copy, "-", &posn);
+#endif
```

Comment:
Similarly, should just use one or the other here.  strtok_s is the [ISO standard](https://en.cppreference.com/w/c/string/byte/strtok).  strtok_r is POSIX..  host.h can #define one to the other as needed, like it does for strdup today.

Reply:
Update per comment. Thanks